pandas/tests/test_infer_and_convert.py

```diff
@@ -102,6 +102,12 @@ def test_scientific_no_exponent(self):
         result = lib.maybe_convert_numeric(arr, set(), False, True)
         self.assertTrue(np.all(np.isnan(result)))
 
+    def test_convert_non_hashable(self):
+        # Test for Bug #13324
+        arr = np.array([[10.0, 2], 1.0, 'apple'])
+        result = lib.maybe_convert_numeric(arr, set(), False, True)
+        np.testing.assert_array_equal(result, np.array([np.nan, 1.0, np.nan]))
```

Comment:
use ``tm.assert_numpy_array_equal`` (we don't use th numpy testing functions at all; in fact there is a check again them; though this one slipped thru)